plugins/node/opentelemetry-instrumentation-pg/src/internal-types.ts

```diff
@@ -46,7 +46,10 @@ export type PgPoolCallback = (
   done: (release?: any) => void
 ) => void;
 
-export type PgErrorCallback = (err: Error) => void;
+// Even though the documented signature for connect() callback is `(err) => void`
+// `pg` actually also passes the client if the connection was successful and some
+// packages(`knex`) might rely on that
+export type PgErrorCallback = (err: Error, client?: pgTypes.Client) => void;
```

Comment:
I wonder if we actually need this type in the instrumentation. Is it used for anything meaningful? Can we replace where it is used with `Function` which signal that we don't care about the arguments of the function?

Reply:
good point, but on the other hand, why shouldn't we always use `Function`? It still gives some readability. Anyway, I removed it in this case since we don't want to be committed to the callback signature.

Reply:
I think we should declare a function type when it has a meaning in the application, e.g. when we depend on the actual parameters or return value type in the source code. Generic `Function`, as I understand it, means that we get a value that is a function and we don't care about its parameters

Reply:
Maybe we could declare a function where the first argument is `err: Error` and then [rest parameters typed as `any`](https://www.typescriptlang.org/docs/handbook/2/functions.html#rest-parameters-and-arguments):

```ts
export type PgErrorCallback = (err: Error, ...otherArguments: any[]) => void;
```

I think this might be the correct typing for the callback function, but feel free to choose whichever syntax you prefer.